### DISCUSSION TITLE
[Blog Plugin] Option to skip/ignore invalid posts

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#5688] [Ideas]
As of right now do I have to edit countless posts to update the date to a proper ISO format date.

This is annoying, especially if my only goal right now is to debug and test out the blog plugin.
It would be very handy to have an option to disable mkdocs material from shutting down the serve if it finds a file that has something wrong with it...

**Comment by squidfunk** [comment]
Thanks for suggesting. Personally, I think the blog plugin failing and telling you what you need to do for it to work again is likely a better way. We might look into auto-populating the date field in the future, possibly by using creation dates from git, but it's nothing that is planned right now.

**Reply by Andre601** [reply]
The main issue with that is, that the actual line telling you the cause is actually burried in countless stacktraces that come right after.

As a normal user without python knowledge would this most likely be overwhelming and make it seem like an actual bug rather than a user-error.

tl;dr: Errors in general should be more human readable.

**Reply by Andre601** [reply]
Also, another very common use case is this:
You currently view your page with `mkdocs serve`. Now you create a new MD file to make a new blog post with, but since the file is empty at first, does Material for MkDocs' blog plugin force MkDocs to shut down because of a missing date, forcing you to fill it with data first before starting again.

**Reply by squidfunk** [reply]
Jup, we've started to make the built-in plugins more resilient, and the optimize and new social plugin already exhibit this behavior – if an error occurs, the rest of the plugin's functionality is just skipped until the next cycle. We'll improve the blog plugin in the coming months to be more resilient as well!

**Reply by Andre601** [reply]
Just found another (more common I would assume) case for why the blog plugin shouldn't stop the serve: Index pages.

In my case do I have sub-folders to sort posts per year. The downside of this is, that a path such as `/2023/` won't have an index page to server, resulting in a 404.
The problem now is, that I can't create an `index.md` page in it to display some info or do something else because the blog plugin expects every single MD file within the posts directory to be a blog post.

And since the index page lacks necessary metadata such as a date does the build fail and the serve cancel.
So an option (Maybe with metadata like `blog-post: false`) to ignore MD files through the blog plugin would be wonderful to have.

**Reply by squidfunk** [reply]
> The problem now is, that I can't create an index.md page in it to display some info or do something else because the blog plugin expects every single MD file within the posts directory to be a blog post.

Just move them out of the `posts` directory.

**Reply by Andre601** [reply]
Not sure how this helps with the main issue?
Like I still would want an index page for the folders... But I use the archive system now to achieve this.

**Reply by squidfunk** [reply]
Everything in posts is considered to be a post. Everything outside is considered a page. We're definitely not planning to add `blog-post: false` to metadata, that's just not a fit for the plugin. However, you can always fork the plugin and add it yourself, once you need it, because that's how Open Source works 😊

**Reply by Andre601** [reply]
So a `docs/2023` folder with an index should work (Given final pages don't have the `/posts/` section in the URL)? 🤔 

I wouldn't know of other ways to do this...

~~Also, to be a bit direct and blunt here: This is just stupid.~~ I aplogize for this bad wording. I should've chosen "confusing" here, as that's what I feel about it. That it is rather puzzling to me to understand.

Having the plugin force you to have only blog posts in one folder without *any* way to ignore one as blog post feels like a seriously bad design.
Perhaps the user wants to have some extra pages there with info?

If the above solution wouldn't work (I doubt it but you never know if there wouldn't be a conflict of some kind) then they had no solution here that wouldn't be a hacky aproach (i.e. move a dummy index page into the generated directory... Tho this might have issues with search indexes and alike).

Also, saying "just fork it" is insulting to me. It's a valid option, sure. But not every person may have the knowledge of the python coding language, nor how this entire repository is structured and designed to make the necessary changes without it having crashes every time.
Again, it's a valid option, but not for everyone.

**Reply by squidfunk** [reply]
> Also, to be a bit direct and blunt here: This is just stupid.

Honestly, use something else. I have nothing else to say.

**Reply by Andre601** [reply]
I apologize for being so rude. I should've worded it a bit differently. It just... confuses me for a lack of a better term as to what the reasons could be for why there isn't a ignore option.
Like I cannot wrap my head around as to why you're against the option to have non-blog post files within the posts directory. Is it too resource intensive to go through the file and apply a different parsing? Is it more complicated?

I would like to understand the reasonings for all this.

I also aplogized to you on Mastodon.
